src/smooch-node.js

```diff
@@ -2,38 +2,11 @@ import { Smooch as SmoochBase } from './smooch';
 import { WebhooksApi } from './api/webhooks';
 import { MenuApi } from './api/menu';
 import * as jwt from './utils/jwt';
+import credential from './utils/credential';
 
 export class Smooch extends SmoochBase {
     constructor(auth = {}, options = {}) {
-        if (auth.keyId || auth.secret) {
-            if (!auth.scope) {
-                throw new Error('Invalid auth: missing scope.');
-            }
-
-            if (!auth.keyId) {
-                throw new Error('Invalid auth: missing keyId.');
-            }
-
-            if (!auth.secret) {
-                throw new Error('Invalid auth: missing secret.');
-            }
-        }
-
-        if (auth.keyId && auth.secret && auth.scope) {
-            const jwtBody = {
-                scope: auth.scope
-            };
-
-            if (auth.userId) {
-                jwtBody.userId = auth.userId;
-            }
-
-            auth = {
-                jwt: jwt.generate(jwtBody, auth.secret, auth.keyId)
```

Comment:
@alavers jwt.generate is the only requiring the jwt lib and it's only used here.